### PR TITLE
Add StyleSheetList[Symbol.iterator] for browsers which have StyleSheetList

### PR DESCRIPTION
**Pull request by JakeChampion**
There is no specification for this feature that I can find, but it does seem to exists in Chrome and Firefox. It does not exist in Safari.

This polyfill currently doesn't target IE8 as IE8 doesn't have `document.styleSheets` and we do not have a polyfill for this as of yet.

### CODE REVIEW

File: polyfills/Object/defineProperties/polyfill.js [hunk 1]
```diff
@@ -1,6 +1,8 @@
 Object.defineProperties = function defineProperties(object, descriptors) {
 	for (var property in descriptors) {
-		Object.defineProperty(object, property, descriptors[property]);
+		if (Object.hasOwnProperty(property)) {
```
Comment (triblondon) [inline polyfills/Object/defineProperties/polyfill.js]: Are you sure about this?  Seems like if anything it should be checking the inverse of this condition.

File: polyfills/StyleSheetList/@@iterator/config.json [hunk 2]
```diff
@@ -0,0 +1,20 @@
+{
+	"browsers": {
+		"ie": "8 <",
```
Comment (triblondon) [inline polyfills/StyleSheetList/@@iterator/config.json]: Uncommon semver syntax - we would commonly write `8 - *`

File: polyfills/StyleSheetList/@@iterator/config.json [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+		"firefox": "*",
+		"chrome": "*",
+		"safari": "*",
+		"android": "*",
+		"ios_saf": "*",
+		"opera": "*"
+	},
+	"dependencies": [
+		"_ArrayIterator",
+		"Symbol.iterator"
```
Comment (triblondon) [inline polyfills/StyleSheetList/@@iterator/config.json]: Does the polyfill work without a `StyleSheetList` polyfill?

File: polyfills/StyleSheetList/@@iterator/detect.js [hunk 4]
```diff
@@ -0,0 +1 @@
+'Symbol' in this && 'iterator' in this.Symbol && !!StyleSheetList.prototype[Symbol.iterator]
```
Comment (triblondon) [inline polyfills/StyleSheetList/@@iterator/detect.js]: I would suggest adding a test for `StyleSheetList` as well.

File: polyfills/StyleSheetList/@@iterator/config.json (hunk 3 above)
Comment (JakeChampion) [inline polyfills/StyleSheetList/@@iterator/config.json]: No it doesn't, IE8 needs a `StyleSheetList` polyfill. I decided to not target IE8 for now.

**Pull request closed without merging**